source/invariant-of.d.ts

```diff
@@ -0,0 +1,72 @@
+declare const tag: unique symbol;
+
+declare type InvariantSignature<Type> = {
+	// Make invariant function signature.
+	readonly [tag]: (argument: Type) => Type;
+};
+
+/**
+Create an [invariant type](https://basarat.gitbook.io/typescript/type-system/type-compatibility#footnote-invariance), which is a type that does not accept supertypes and subtypes.
+
+Use-case:
+- Prevent runtime errors that may occur due to assigning subtypes to supertypes.
+- Improve type signature of object methods like [`Object.keys()` or `Object.entries()`](https://github.com/microsoft/TypeScript/pull/12253#issuecomment-263132208) by sealing the object type.
```

Comment:
Can you think of any more specific use-cases?

Reply:
It's hard to think of any more specific use-cases for me.